core/lib/src/response/response.rs

```diff
@@ -807,6 +816,26 @@ impl<'r> Response<'r> {
         self.body = Body::with_unsized(body);
     }
 
+    /// Returns a instance of the `Upgrade`-trait when the `Response` is upgradeable
+    #[inline(always)]
+    pub fn upgrade(&self) -> &Option<Box<dyn Upgrade<'static> + Send>> {
+        &self.upgrade
+    }
```

Comment:
```suggestion
    pub fn upgrade(&self) -> Option<&Box<dyn Upgrade<'static> + Send>> {
        self.upgrade.as_ref()
    }
```
It's more idiomatic to return `Option<&T>` instead of `&Option<T>`.

Reply:
Should `upgrade_mut` also be changed to `fn upgrade_mut(&mut self) -> Option<&mut Box<dyn Upgrade<'static> + Send>>`? If I do that via `as_mut()`, the same issue as in your other comment arises: `send_response()` moves out the response.

Could work with a new method:
```rust
pub fn take_upgrade(&mut self) -> Option<Box<dyn Upgrade<'static> + Send>> {
    self.upgrade.take()
}
```